consensus/tests/history_sync.rs

```diff
@@ -0,0 +1,406 @@
+mod sync_utils;
+
+use std::path::PathBuf;
+use std::sync::Arc;
+use std::time::Duration;
+
+use futures::StreamExt;
+use nimiq_blockchain_proxy::BlockchainProxy;
+use nimiq_bls::cache::PublicKeyCache;
+use nimiq_consensus::sync::syncer_proxy::SyncerProxy;
+use nimiq_test_log::test;
+use nimiq_test_utils::node::TESTING_BLS_CACHE_MAX_CAPACITY;
+use nimiq_test_utils::zkp_test_data::{zkp_test_exe, KEYS_PATH};
+use nimiq_zkp_component::ZKPComponent;
+use parking_lot::{Mutex, RwLock};
+
+use nimiq_block_production::BlockProducer;
+use nimiq_blockchain::{Blockchain, BlockchainConfig};
+use nimiq_blockchain_interface::AbstractBlockchain;
+use nimiq_consensus::consensus::Consensus;
+use nimiq_consensus::sync::history::{cluster::SyncCluster, HistoryMacroSync};
+use nimiq_database::volatile::VolatileEnvironment;
+use nimiq_genesis::NetworkId;
+use nimiq_network_interface::network::Network as NetworkInterface;
+use nimiq_network_libp2p::Network;
+use nimiq_network_mock::MockHub;
+use nimiq_primitives::policy::Policy;
+use nimiq_test_utils::{
+    blockchain::{produce_macro_blocks, signing_key, voting_key},
+    test_network::TestNetwork,
+};
+use nimiq_utils::time::OffsetTime;
+
+use crate::sync_utils::{sync_two_peers, SyncMode};
+
+#[test(tokio::test)]
+async fn two_peers_can_sync_empty_chain() {
+    sync_two_peers(0, 0, SyncMode::History).await
+}
+
+#[test(tokio::test)]
+async fn two_peers_can_sync_single_batch() {
+    sync_two_peers(1, 0, SyncMode::History).await
+}
+
+#[test(tokio::test)]
+async fn two_peers_can_sync_two_batches() {
+    sync_two_peers(2, 0, SyncMode::History).await
+}
+
+#[test(tokio::test)]
+async fn two_peers_can_sync_epoch_minus_batch() {
+    sync_two_peers(
+        (Policy::batches_per_epoch() - 1) as usize,
+        0,
+        SyncMode::History,
+    )
+    .await
+}
+
+#[test(tokio::test)]
+async fn two_peers_can_sync_epoch_plus_batch() {
+    sync_two_peers(
+        (Policy::batches_per_epoch() + 1) as usize,
+        0,
+        SyncMode::History,
+    )
+    .await
+}
+
+#[test(tokio::test)]
+async fn two_peers_can_sync_epoch_plus_two_batches() {
+    sync_two_peers(
+        (Policy::batches_per_epoch() + 2) as usize,
+        0,
+        SyncMode::History,
+    )
+    .await
+}
+
+#[test(tokio::test)]
+async fn two_peers_can_sync_single_epoch() {
+    sync_two_peers(Policy::batches_per_epoch() as usize, 0, SyncMode::History).await
+}
+
+#[test(tokio::test)]
+async fn two_peers_can_sync_two_epochs() {
+    sync_two_peers(
+        (Policy::batches_per_epoch() * 2) as usize,
+        0,
+        SyncMode::History,
+    )
+    .await
+}
+
+#[test(tokio::test)]
+#[ignore]
```

Comment:
I'd either mark it as `ignore` or comment the test content

Reply:
It was like this before, we just moved it